amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/WebApiDeclarations.scala

```diff
@@ -33,70 +33,53 @@ import amf.shapes.client.scala.model.domain.Example
 import amf.shapes.internal.domain.metamodel.CreativeWorkModel
 import amf.shapes.internal.spec.common.error.ErrorNamedExample
+import amf.shapes.internal.spec.common.parser.ShapeDeclarations
 import org.yaml.model.{YNode, YPart}
 
 /** Declarations object.
   */
 class WebApiDeclarations(
-    val alias: Option[String],
-    var libs: Map[String, WebApiDeclarations] =
-      Map(), // TODO: sync this with libraries, is confusing and we end having tw different maps (one of them is not being filled)
-    var frags: Map[String, FragmentRef] = Map(),
-    var shapes: Map[String, Shape] = Map(),
-    var anns: Map[String, CustomDomainProperty] = Map(),
-    var resourceTypes: Map[String, ResourceType] = Map(),
-    var parameters: Map[String, Parameter] = Map(),
-    var payloads: Map[String, Payload] = Map(),
-    var traits: Map[String, Trait] = Map(),
-    var securitySchemes: Map[String, SecurityScheme] = Map(),
-    var responses: Map[String, Response] = Map(),
-    var examples: Map[String, Example] = Map(),
-    var requests: Map[String, Request] = Map(),
-    var headers: Map[String, Parameter] = Map(),
-    var links: Map[String, TemplatedLink] = Map(),
-    var correlationIds: Map[String, CorrelationId] = Map(),
-    var callbacks: Map[String, List[Callback]] = Map(),
-    var messages: Map[String, Message] = Map(),
-    var messageBindings: Map[String, MessageBindings] = Map(),
-    var operationBindings: Map[String, OperationBindings] = Map(),
-    var channelBindings: Map[String, ChannelBindings] = Map(),
-    var serverBindings: Map[String, ServerBindings] = Map(),
-    var operationTraits: Map[String, Operation] = Map(),
-    var messageTraits: Map[String, Message] = Map(),
+    alias: Option[String],
     val errorHandler: AMFErrorHandler,
-    val futureDeclarations: FutureDeclarations,
-    var others: Map[String, BaseUnit] = Map(),
-    var extensions: Map[String, Dialect] = Map(),
-    var shapeExtensions: Map[String, Seq[Shape]] = Map() // we can have 0+ shape extensions with the same name
-) extends Declarations(libs, frags, anns, errorHandler, futureDeclarations = futureDeclarations) {
-
-  def promoteExternalToDataTypeFragment(text: String, fullRef: String, shape: Shape): Unit = {
-    fragments.get(text) match {
-      case Some(fragmentRef) =>
-        promotedFragments :+= DataTypeFragment()
-          .withId(fragmentRef.location.getOrElse(fullRef))
-          .withLocation(fragmentRef.location.getOrElse(fullRef))
-          .withEncodes(shape)
-        fragments += (text -> FragmentRef(shape, fragmentRef.location))
-      case _ =>
-        promotedFragments :+= DataTypeFragment().withId(fullRef).withLocation(fullRef).withEncodes(shape)
-        fragments += (text -> FragmentRef(shape, None))
-    }
+    val futureDeclarations: FutureDeclarations
+) extends ShapeDeclarations(alias, errorHandler, futureDeclarations = futureDeclarations) {
+
+  var resourceTypes: Map[String, ResourceType]          = Map()
```

Comment:
The main reason had to move them is because when adding the intermediate `ShapeDeclarations` class the scala compiler threw an error regarding vars and constructor parameters.

Another reason is that we were duplicating variable declarations for the same things. For example we had two variables for libraries (libs and libraries), two for fragments (frags and fragments) and so on. Moving those variables from the class signature helped me unify things.

Besides, when I saw it there wasn't any class that was using the constructor with all the parameters. The only constructor params being used was the one I left. When classes populated a Declarations instance they did it by copying field by field